analyzers/src/SonarAnalyzer.CSharp/Rules/MethodParameterUnused.cs

```diff
@@ -256,15 +256,23 @@ private class MethodContext
         {
             public readonly SonarSyntaxNodeReportingContext Context;
             public readonly IMethodSymbol Symbol;
+            public readonly BaseMethodDeclarationSyntax MethodDeclaration;
+            public readonly LocalFunctionStatementSyntaxWrapper LocalFunctionDeclaration;
             public readonly ParameterListSyntax ParameterList;
             public readonly BlockSyntax Body;
             public readonly ArrowExpressionClauseSyntax ExpressionBody;
 
             public MethodContext(SonarSyntaxNodeReportingContext context, BaseMethodDeclarationSyntax declaration)
-                : this(context, declaration.ParameterList, declaration.Body, declaration.ExpressionBody()) { }
+                : this(context, declaration.ParameterList, declaration.Body, declaration.ExpressionBody())
+            {
+                MethodDeclaration = declaration;
+            }
```

Comment:
Should be arrow method. Same below

Reply:
code removed - comment not relevant anymore.